test/e2e/operator_test.go

```diff
@@ -1462,6 +1463,192 @@ func TestNetworkLoadBalancer(t *testing.T) {
+		},
+	}
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoService); err != nil {
+			t.Fatalf("failed to delete service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+		}
+	}()
+
+	echoRoute := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      echoPod.Name,
+			Namespace: echoPod.Namespace,
+		},
+		Spec: routev1.RouteSpec{
+			Host: "unique-id-header-test.com",
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: echoService.Name,
+			},
+		},
+	}
+	if err := kclient.Create(context.TODO(), echoRoute); err != nil {
+		t.Fatalf("failed to create route %s/%s: %v", echoRoute.Namespace, echoRoute.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), echoRoute); err != nil {
+			t.Fatalf("failed to delete route %s/%s: %v", echoRoute.Namespace, echoRoute.Name, err)
+		}
+	}()
+
+	clientPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "unique-id-header-test",
+			Namespace: echoRoute.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "curl",
+					Image:   deployment.Spec.Template.Spec.Containers[0].Image,
+					Command: []string{"/bin/curl"},
+					Args: []string{
+						"-s",
+						"--retry", "300",
+						"--retry-delay", "1",
+						"--max-time", "2",
+						"--resolve",
+						echoRoute.Spec.Host + ":80:" + service.Spec.ClusterIP,
+						"http://" + echoRoute.Spec.Host,
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyNever,
+		},
+	}
+	if err := kclient.Create(context.TODO(), clientPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+	}
+	defer func() {
+		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
+			t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+		}
+	}()
+
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		t.Fatalf("failed to get kube config: %v", err)
+	}
+	client, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatalf("failed to create kube client: %v", err)
+	}
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (bool, error) {
+		readCloser, err := client.CoreV1().Pods(clientPod.Namespace).GetLogs(clientPod.Name, &corev1.PodLogOptions{
+			Container: "curl",
+			Follow:    false,
+		}).Stream(context.TODO())
+		if err != nil {
+			t.Logf("failed to read output from pod %s: %v", clientPod.Name, err)
+			return false, nil
+		}
+		scanner := bufio.NewScanner(readCloser)
+		defer func() {
+			if err := readCloser.Close(); err != nil {
+				t.Errorf("failed to close reader for pod %s: %v", clientPod.Name, err)
+			}
+		}()
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasPrefix(line, "x-unique-id:") {
```

Comment:
Would it be worth enhancing this test to check 2 different requests so the uniqueness of the unique-id header could be verified? I may be overthinking this, but it seems that there would be some benefit in verifying the default unique-id header format string works as intended (as far as the unique part goes). Thoughts?

Reply:
Done.  Thanks for the suggestion!